### PR TITLE
SYNCT-54: SyncClient: Push to the Master through FHIR

### PR DESCRIPTION
**Pull request by ubap**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java [hunk 1]
```diff
@@ -23,32 +23,36 @@
...
+        IBaseResource baseResource = (IBaseResource) object;
         try {
-            result = parser.parseResource(classType, stringObject);
-        } catch(DataFormatException e) {
-            log.error(String.format("Could not parse String to Object: %s", stringObject));
+            return restTemplate.postForEntity(category, baseResource, String.class);
+        } catch(RestClientException e) {
+            log.error(String.format("Exception occurred when posting object. Category: %s, url: %s", category, url),
+                    e);
+            throw e;
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java]: Log & (Re)throw anti-pattern

File: api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java [hunk 2]
```diff
@@ -23,32 +23,36 @@
...
+            throws RestClientException {
         prepareRestTemplate(username, password);
-        String stringObject = "";
         try {
-            stringObject = restTemplate.getForObject(url, resolveCategory(category)).toString();
-        } catch(HttpClientErrorException e) {
-            log.error(String.format("Resource %s not found", category));
+            return restTemplate.getForObject(url, resolveCategory(category));
+        } catch(RestClientException e) {
+            log.error(String.format("Exception occurred when getting object. Category: %s, url: %s", category, url),
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java]: Log & (Re)throw anti-pattern

File: api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java [hunk 3]
```diff
@@ -23,32 +23,24 @@
...
-        } catch(DataFormatException e) {
-            log.error(String.format("Could not parse String to Object: %s", stringObject));
-        }
-        return result;
+    @Override
+    public ResponseEntity<String> postObject(String category, String url, String username, String password, Object object)
+            throws RestClientException {
+        prepareRestTemplate(username, password);
+        IBaseResource baseResource = (IBaseResource) object;
+        return restTemplate.postForEntity(category, baseResource, String.class);
```
Comment (kcissewski) [inline api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java]: I think that first parameter should be URL not category.

Comment (ubap) [reply api/src/main/java/org/openmrs/module/fhir/api/client/FHIRClient.java]: Thanks a lot! Thats my mislook

**Pull request merged**